src/server/master_server/master_metadata_service_impl.h

```diff
@@ -14,16 +17,22 @@ namespace service {
 class MasterMetadataServiceImpl final
     : public protos::grpc::MasterMetadataService::Service {
  public:
+  MasterMetadataServiceImpl(common::ConfigManager* config_manager) : 
```

Comment:
nit: why this? you can get config manager by doing ConfigManager::GetConfig(). It's singleton. Am I missing something here?

Reply:
Actually I wrote ConfigManager NOT as a singleton; this is just a static method; each time you call GetConfig it parses a new file again

Reply:
Thanks for clarifying Gan, correct, because GetConfig is not a singleton (as it takes a file path as input), we need a reference of config manager here.